### ISSUE TITLE
False positive reportPossiblyUnboundVariable after pattern matching, inside a loop

### ISSUE THREAD

**Issue by michelbl** [microsoft/pyright#7371]
**Describe the bug**
On the following code, pyright reports the error `error: "v" is possibly unbound (reportPossiblyUnboundVariable)`, while it should not.

**Code or Screenshots**
```python
from typing import Literal


def foo(lit: Literal["a", "b"]) -> None:
    for _ in range(2):
        match lit:
            case "a":
                v = "123"

            case "b":
                v = "234"

        v.replace(",", ".")
```

**VS Code extension or command-line**

pyright CLI 1.1.352

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.
